build/dsc.rake

```diff
@@ -67,16 +72,61 @@ eod
         FileUtils.cp_r "#{dsc_resources_path}/.", "#{dsc_resources_path_tmp}/"
       end
 
-      FileUtils.rm_rf(Dir["#{dsc_resources_path_tmp}/**/.git"])
-
       blacklist = ['xChrome', 'xDSCResourceDesigner', 'xDscDiagnostics',
                    'xFireFox', 'xSafeHarbor', 'xSystemSecurity']
       puts "Cleaning out black-listed DSC resources: #{blacklist}"
       blacklist.each { |res| FileUtils.rm_rf("#{dsc_resources_path_tmp}/xDSCResources/#{res}") }
 
+      resource_tags = {}
+      resource_tags = YAML::load_file("#{dsc_resources_file}") if File.exist? dsc_resources_file
+
+      puts "Getting latest release tags for DSC resources..."
+      Dir["#{dsc_resources_path_tmp}/xDSCResources/*"].each do |dsc_resource_path|
+        dsc_resource_name = Pathname.new(dsc_resource_path).basename
+        FileUtils.cd(dsc_resource_path) do
+          # --date-order probably doesn't matter
+          tags_raw = %x{ git log --tags --pretty=format:'%D' --simplify-by-decoration --date-order }
+          # If the conversion of string to version starts to result in errors,
+          # we should explore pushing this out out to a method where we can
+          # clean up the tags that may have prerelease versions in them
+          # similar to what was done in the Chocolatey module
+          versions = tags_raw.scan(/(\S+)\-PSGallery/).map { | ver | Gem::Version.new(ver[0]) }
+          if versions.empty?
+            raise "#{dsc_resource_name} does not have any '*-PSGallery' tags. Appears it has not been released yet. Tags found #{tags_raw.to_s}"
+          end
+
+          latest_version = versions.max.to_s + "-PSGallery"
+          tracked_version = resource_tags["#{dsc_resource_name}"]
+
+          update_version = update_versions
+          update_version = true if tracked_version.nil?
+
+          if update_version
+            puts "Latest/available tag for #{dsc_resource_name} is #{latest_version}. Switching to that tag."
+            checkout_version = latest_version
+          else
+            puts "Using tag for #{dsc_resource_name} is #{tracked_version}. Switching to that tag."
```

Comment:
We also need to update this to remove the word tag as it could just be a sha. @glennsarti are you updating these?